rlmolecule/mcts/mcts.py

```diff
@@ -113,12 +114,17 @@ def _expand(self, leaf: MCTSVertex) -> None:
         """
         Expansion step of MCTS
         From Wikipedia (https://en.wikipedia.org/wiki/Monte_Carlo_tree_search):
-        Expansion: Unless L ends the game decisively (e.g. win/loss/draw) for either player, create one (or more) child
-        vertices and choose vertex C from one of them. Child vertices are any valid moves from the game position defined by L.
+        Expansion: Unless L ends the game decisively (e.g. win/loss/draw) for either player, create one (or more)
+        child vertices and choose vertex C from one of them. Child vertices are any valid moves from the game
+        position defined by L.
         """
         if leaf.children is None:
             leaf.children = [self.get_vertex_for_state(state) for state in leaf.state.get_next_actions()]
 
+            for child in leaf.children:
+                if child.children is not None:
```

Comment:
Can you add a comment here or in dfs() that only the already visited nodes will be checked here?